OSPSuite.Presentation.Importer/Services/IImporter.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using DevExpress.Utils.Extensions;
+using OSPSuite.Core.Importer;
+using OSPSuite.Presentation.Importer.Core;
+using IoC = OSPSuite.Utility.Container.IContainer;
+
+namespace OSPSuite.Presentation.Importer.Services
+{
+   public interface IImporter
+   {
+      IDataSourceFile LoadFile(IReadOnlyList<ColumnInfo> columnInfos, string fileName);
+      void AddFromFile(IDataFormat format, IReadOnlyDictionary<string, IDataSheet> dataSheets, IReadOnlyList<ColumnInfo> columnInfos, IDataSource alreadyExisting);
+      IEnumerable<IDataFormat> AvailableFormats(IUnformattedData data, IReadOnlyList<ColumnInfo> columnInfos);
+
+      IEnumerable<string> NamesFromConvention
+      (
+         string namingConvention,
+         string fileName,
+         IDictionary<string, IDataSet> dataSets,
+         IEnumerable<MetaDataMappingConverter> mappings
+      );
+   }
+
+   public class Importer : IImporter
+   {
+      private readonly IoC _container;
+      private readonly IDataSourceFileParser _parser;
+
+      public Importer( IoC container, IDataSourceFileParser parser)
+      {
+         _container = container;
+         _parser = parser;
+      }
+
+      public IEnumerable<IDataFormat> AvailableFormats(IUnformattedData data, IReadOnlyList<ColumnInfo> columnInfos)
+      {
+         return _container.ResolveAll<IDataFormat>()
```

Comment:
use IRepositor<> it does it for you automatically . No need to use container

Reply:
We re not exactly sure of what is meant here or how to use IRepository<>